src/Main.elm

```diff
@@ -628,21 +653,108 @@ raisedDiv =
         ]
 
 
-modalView : Model -> Html Msg
-modalView model =
+type Menu
+    = DifficultyMenu Difficulty GridProperties
+
+
+type MenuMsg
+    = SetDifficulty Difficulty
+
+
+updateMenu : MenuMsg -> Menu -> Menu
+updateMenu msg (DifficultyMenu difficulty fields) =
+    case msg of
+        SetDifficulty (Custom properties) ->
+            DifficultyMenu (Custom properties) properties
+
+        SetDifficulty d ->
+            DifficultyMenu d fields
```

Comment:
i like this because, up in our Main`, we have a `Maybe Menu`. i don't want to deal with that uncertainty once i'm "in" the menu. this is a more definite function, going `Menu -> Menu`. up above, i use `Maybe.map` to deal with the uncertainty.